From: Steve French <smfrench@gmail.com>
Date: Thu, 25 Jan 2018 09:28:25 -0600
Subject: [PATCH] move a few externs to smbdirect.h to eliminate warning
Git-commit: 6b314714ff6337465a730195952e1b1f10d97063
Patch-mainline: v4.15
References: bsc#1144333

Quiet minor sparse warnings in new SMB3 rdma patch series
("symbol was not declared ...") by moving these externs to smbdirect.h

Signed-off-by: Steve French <smfrench@gmail.com>
Reviewed-by: Ronnie Sahlberg <lsahlber@redhat.com>
Acked-by: Aurelien Aptel <aaptel@suse.com>
---
 fs/cifs/cifs_debug.c | 9 ---------
 fs/cifs/smbdirect.h  | 9 +++++++++
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/fs/cifs/cifs_debug.c b/fs/cifs/cifs_debug.c
index f491340f32ad..c7a863219fa3 100644
--- a/fs/cifs/cifs_debug.c
+++ b/fs/cifs/cifs_debug.c
@@ -483,15 +483,6 @@ static const struct file_operations cifs_##name##_proc_fops = { \
 	.write		= name##_write, \
 }
 
-extern int rdma_readwrite_threshold;
-extern int smbd_max_frmr_depth;
-extern int smbd_keep_alive_interval;
-extern int smbd_max_receive_size;
-extern int smbd_max_fragmented_recv_size;
-extern int smbd_max_send_size;
-extern int smbd_send_credit_target;
-extern int smbd_receive_credit_max;
-
 PROC_FILE_DEFINE(rdma_readwrite_threshold);
 PROC_FILE_DEFINE(smbd_max_frmr_depth);
 PROC_FILE_DEFINE(smbd_keep_alive_interval);
diff --git a/fs/cifs/smbdirect.h b/fs/cifs/smbdirect.h
index fdb8df84fa1f..f9038daea194 100644
--- a/fs/cifs/smbdirect.h
+++ b/fs/cifs/smbdirect.h
@@ -24,6 +24,15 @@
 #include <rdma/rdma_cm.h>
 #include <linux/mempool.h>
 
+extern int rdma_readwrite_threshold;
+extern int smbd_max_frmr_depth;
+extern int smbd_keep_alive_interval;
+extern int smbd_max_receive_size;
+extern int smbd_max_fragmented_recv_size;
+extern int smbd_max_send_size;
+extern int smbd_send_credit_target;
+extern int smbd_receive_credit_max;
+
 enum keep_alive_status {
 	KEEP_ALIVE_NONE,
 	KEEP_ALIVE_PENDING,
-- 
2.16.4

